grails-app/controllers/au/org/emii/aatams/SensorController.groovy

```diff
@@ -32,8 +32,6 @@ class SensorController extends ReportController {
         def tag = tagFactoryService.lookupOrCreate(params.tag)
         assert(tag)
 
-        params.tag = tag
```

Comment:
It would be really good to see this code covered with an integration test - to lock-down the correct behavior at the database level - in terms of the underlying foreign key references and sequence generation. 

It's very difficult to understand what's going on, by simply looking at the surface gorm code. Also this code appears to have diverged significantly from the automatically generated grails boilerplate.

At the same time, @pblain probably will want to see this completed by the end of the iteration.

@pblain thoughts?